presto-main/src/test/java/com/facebook/presto/operator/scalar/TestBitwiseFunctions.java

```diff
@@ -0,0 +1,82 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+
+public class TestBitwiseFunctions
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testBitCount()
+    {
+        assertFunction("bit_count(0, 64)", BIGINT, 0L);
+        assertFunction("bit_count(7, 64)", BIGINT, 3L);
+        assertFunction("bit_count(24, 64)", BIGINT, 2L);
+        assertFunction("bit_count(-8, 64)", BIGINT, 61L);
+        assertFunction("bit_count(" + Integer.MAX_VALUE + ", 64)", BIGINT, 31L);
+        assertFunction("bit_count(" + Integer.MIN_VALUE + ", 64)", BIGINT, 33L);
+        assertFunction("bit_count(" + Long.MAX_VALUE + ", 64)", BIGINT, 63L);
+        assertFunction("bit_count(-" + Long.MAX_VALUE + "-1, 64)", BIGINT, 1L); // bit_count(MIN_VALUE, 64)
+
+        assertFunction("bit_count(0, 32)", BIGINT, 0L);
+        assertFunction("bit_count(7, 32)", BIGINT, 3L);
+        assertFunction("bit_count(24, 32)", BIGINT, 2L);
+        assertFunction("bit_count(-8, 32)", BIGINT, 29L);
+        assertFunction("bit_count(" + Integer.MAX_VALUE + ", 32)", BIGINT, 31L);
+        assertFunction("bit_count(" + Integer.MIN_VALUE + ", 32)", BIGINT, 1L);
+        assertFunction("bit_count(" + Long.MAX_VALUE + ", 32)", BIGINT, 31L);
+        assertFunction("bit_count(-" + Long.MAX_VALUE + "-1, 32)", BIGINT, 1L); // bit_count(MIN_VALUE, 64)
+    }
+
+    @Test
+    public void testBitwiseNot()
+    {
+        assertFunction("bitwise_not(0)", BIGINT, -1L);
+        assertFunction("bitwise_not(-1)", BIGINT, 0L);
+        assertFunction("bitwise_not(8)", BIGINT, -9L);
+        assertFunction("bitwise_not(-8)", BIGINT, 7L);
+        assertFunction("bitwise_not(" + Long.MAX_VALUE + ")", BIGINT, Long.MIN_VALUE);
+        assertFunction("bitwise_not(-" + Long.MAX_VALUE + "-1)", BIGINT, Long.MAX_VALUE); // bitwise_not(MIN_VALUE)
+    }
+
+    @Test
+    public void testBitwiseAnd()
+    {
+        assertFunction("bitwise_and(0, -1)", BIGINT, 0L);
+        assertFunction("bitwise_and(3, 8)", BIGINT, 0L);
+        assertFunction("bitwise_and(-4, 12)", BIGINT, 12L);
+        assertFunction("bitwise_and(60, 21)", BIGINT, 20L);
+    }
+
+    @Test
+    public void testBitwiseOr()
+    {
+        assertFunction("bitwise_or(0, -1)", BIGINT, -1L);
+        assertFunction("bitwise_or(3, 8)", BIGINT, 11L);
+        assertFunction("bitwise_or(-4, 12)", BIGINT, -4L);
+        assertFunction("bitwise_or(60, 21)", BIGINT, 61L);
+    }
+
+    @Test
+    public void testBitwiseXor()
+    {
+        assertFunction("bitwise_xor(0, -1)", BIGINT, -1L);
+        assertFunction("bitwise_xor(3, 8)", BIGINT, 11L);
+        assertFunction("bitwise_xor(-4, 12)", BIGINT, -16L);
+        assertFunction("bitwise_xor(60, 21)", BIGINT, 41L);
```

Comment:
Instead of having `41L` as a constant, it would be more obviously correct if you put `60L ^ 21L`. Same for the other tests